pulsar-metadata/src/main/java/org/apache/pulsar/metadata/impl/ZKMetadataStore.java

```diff
@@ -73,13 +74,15 @@ public ZKMetadataStore(String metadataURL, MetadataStoreConfig metadataStoreConf
                     .allowReadOnlyMode(metadataStoreConfig.isAllowReadOnlyOperations())
                     .sessionTimeoutMs(metadataStoreConfig.getSessionTimeoutMillis())
                     .watchers(Collections.singleton(event -> {
```

Comment:
if enableSessionWatcher is false we can skip adding the watcher at all, can't we ?

Reply:
There's a bit of a chicken and egg problem there that would make the change a bit uglier. The ZKSessionWatcher needs to have the already build ZK client object, while we need to pass the list of watchers to construct the ZK client.

That means that when we build  the ZK client we still don't have the set of watchers (that is also happening in the current implementation, which is why the field is not `final` and we're reading the variable out. 

In practice, I don't think it makes any difference to have the watcher that just ignores the session event instead of no watcher.

Reply:
Okay